runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -340,9 +340,9 @@ void J9::RecognizedCallTransformer::process_java_lang_StringLatin1_inflate_BIBII
 #if defined(OMR_GC_SPARSE_HEAP_ALLOCATION)
    if (TR::Compiler->om.isOffHeapAllocationEnabled())
       {
-      dstOff = TR::TransformUtil::generateConvertArrayElementIndexToOffsetTrees(comp(), dstOff, strideNode, 0, false);
+      TR::Node *tmpNode = TR::TransformUtil::generateConvertArrayElementIndexToOffsetTrees(comp(), dstOff, strideNode, 0, false);
       srcAddr = TR::TransformUtil::generateArrayElementAddressTrees(comp(), srcObj, srcOff);
```

Comment:
Unless I'm misreading, I think `srcOff` will be of type `TR::Int32` here, but the call to [TR::TransformUtil::generateArrayElementAddressTrees](https://github.com/eclipse-openj9/openj9-omr/blob/f11383719fe05b1447239b9c07ab10d1f93dea47/compiler/optimizer/OMRTransformUtil.cpp#L471) will use it in an `aladd` operation.  Does this need a call to `TR::TransformUtil::generateConvertArrayElementIndexToOffsetTrees` in the same way as was done for `dstOff`, placing the resultant node in another temporary variable?

If you do that, I think that this code that's currently guarded by `om.isOffHeapAllocationEnabled()` could be used unconditionally, removing the code that's currently in the `else` branch.

Reply:
Right, it seems like we need a `i2l` inserted by `generateArrayElementAddressTrees` in such cases. I would expect that this happens frequently currently but never caused an issue...
@VermaSh do you agree with the change? I can do the coding.


> If you do that, I think that this code that's currently guarded by om.isOffHeapAllocationEnabled() could be used unconditionally, removing the code that's currently in the else branch.
Yup, sounds like a good cleanup.

Reply:
Actually I think keeping `generateArrayElementAddressTrees` as is and calling `generateConvertArrayElementIndexToOffsetTrees` for the source too would be a better solution, as `generateArrayElementAddressTrees` expects an offset and for OffHeap its always 64bit hence long typed, which to me sounds like a fair assumption.

Reply:
I agree with calling `generateConvertArrayElementIndexToOffsetTrees` for source offset node here.